### ISSUE TITLE
Module not working

### ISSUE THREAD

**Issue by aacmsu** [DustinBryant/MMM-Ring#4]
When ringing the bell the stream does not display. Have not changed any configs. 

Did a git pull for any updates but everything seems up to date.

Please assist.

**Comment by DustinBryant** [comment]
I need a little more information.

1. Do you have an active Ring subscription (must have an active Ring subscription)?
2. What version of the MagicMirror are you running?
3. How are you running MagicMirror (Normal, Server Only, Client Only, or Docker)? 
3. Are there any errors posted to the terminal (with MagicMirror running, minimize it [Alt, then Ctrl + 
 m], and look at the terminal from which you started MagicMirror)?
4. Are there any errors on the dev console (with the terminal, navigate to the MagicMirror folder and run ```npm start dev```) 
5. Can you provide your config for MMM-Ring (with username/password redacted)
6. With MagicMirror running, minimize it (with MagicMirror running, minimize it [Alt, then Ctrl + 
 m]), open up file explorer and navigate to the modules/MMM-Ring/public folder. Ring your doorbell. Are .ts files being created?

**Comment by aacmsu** [comment]
1. I do not have a subscription but it was working for the almost month and a half.
2. vs 2.7.1
3. running Normal
4. No errors in terminal
5. running npm start dev
npm ERR! path /home/pi/package.json
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall open
npm ERR! enoent ENOENT: no such file or directory, open '/home/pi/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2019-11-26T16_02_35_594Z-debug.log
6.  {
      module: 'MMM-Ring',
      position: 'middle_center',
      config: {
      ringEmail: '',
      ringPwd: '',
      ringMinutesToStreamVideo: 1,
      ringVideoWidth: '800'
      }
    },
7. No .ts files being created

**Comment by DustinBryant** [comment]
Before I look further, you automatically get a 30 day trial of Ring Protect (ring subscription mentioned in 1) when you purchase a new Ring product. Was your device newly purchased and set up 30 days ago when it was working?

**Comment by aacmsu** [comment]
I have had this Ring doorbell for about 8 months now.

**Comment by DustinBryant** [comment]
It's interesting you say it has been working for over a month but now isn't.

1. Are you able to get normal functionality of Ring either on their website or their mobile app?
2. Have you tried completely removing MMM-Ring and re-installing?

**Comment by aacmsu** [comment]
I have normal functionality in the website as well as mobile app.
I removed the Ring module and reinstalled, but the issue still persists.

**Comment by DustinBryant** [comment]
I started having issues a day ago. I'd recently updated my Raspberry Pi 4's firmware and updated packages and the issue began. It looks to be like an issue with the ring-client-api being used in this module so I have created an issue on the developer's github.

https://github.com/dgreif/ring/issues/194

I'll let you know when/if this issue gets resolved.

**Comment by DustinBryant** [comment]
@aacmsu it seems the unofficial ring api has been fixed. I did testing on my end with updating it and everything seems to be working as usual. Please give it a shot and tell me if this fixes your issue as well.

Navigate to the MMM-Ring module and run
```git pull && npm install```

**Comment by aacmsu** [comment]
This seems to have fixed the issue, though the feed is very delayed. I also notice I am unable to use the api with 2fa.

**Comment by DustinBryant** [comment]
@aacmsu glad it started working for you again. I'm not sure we can really address the delay issue. Not sure if this has happened to you but I have also found the Ring app on android has recently started to become unreliable, nearly broken for live events, and has choppy/delayed video playback.

As for the 2fa, I just updated the [README](https://github.com/DustinBryant/MMM-Ring/blob/master/README.md) and also added [Two Factor Auth (ring2faRefreshToken config option)](https://github.com/DustinBryant/MMM-Ring/wiki/Two-Factor-Auth-(ring2faRefreshToken-config-option)) wiki entry on how to set this up properly.

Check it out and let me know if this works for you.
